### ISSUE TITLE
provider needs to pass the GPU model to the bid price script _before_ it Bids so it can price it (when the GPU `model` is not set in SDL)

### ISSUE THREAD

**Issue by andy108369** [akash-network/support#166]
akash node/network 0.30.0, and 0.32.2
provider-services version 0.4.8, and 0.5.13

**Update (Apr 30, 2024):**
**TL;DR** The issue is basically this:
- `model: *` set in the SDL eventually gets through the bid engine which then gets passed to the bid price script (or whatever reads the `stdin` to process the `resources`);  What's expected is that the actually available GPU which got selected by the K8s engine.
----

The current version of the bid price script [automatically sets](https://github.com/akash-network/support/issues/139#issuecomment-1793078953) the highest price when a specific GPU is not designated in the SDL. This approach is adopted because the script is unable to determine which GPU is being requested or which one is available; therefore, it defaults to the highest price. This serves as a temporary solution to ensure that the provider does not inadvertently offer a high-end GPU at the cost of a lower-end model.
However, it causes an issue when only the lower-end models are available and the client does not explicitly specify the model in the SDL - this way he gets lower-end models for the highest price of the highest-end model.

Provider needs to pass the GPU model it picked to the bid price script _before_ it Bids so it can price it (when the GPU `model` is not set in SDL deployment manifest).

**Comment by andy108369** [comment]
the best provider to test this now is `provider.medc1.com` (`akash1ffpcy473xqs37yvv4jhhh2hsuv786nzs4xt0dj`) which got `t4`, `a100` and `rtx4090` (`a40` probably soon)

**Comment by anilmurty** [comment]
Thanks Andrey - I ran into this yesterday and wasn't aware of this issue - so, appreciated! - going to addd this to backlog
